src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.Unix.cs

```diff
@@ -175,28 +175,33 @@ public bool MoveNext()
 
         private unsafe void FindNextEntry()
         {
-            Span<byte> buffer = _entryBuffer == null ? Span<byte>.Empty : new Span<byte>(_entryBuffer);
-            int result = Interop.Sys.ReadDir(_directoryHandle, buffer, ref _entry);
-            switch (result)
+            if (_entryBuffer == null)
+                return;
```

Comment:
Not sure about this. The old code used `Span<byte>.Empty` in case of null _entryBuffer. Then that empty span was used in ReadDir where it could crash with NullReferenceException because of `ref MemoryMarshal.GetReference`

Reply:
> Then that empty span was used in ReadDir where it could crash with NullReferenceException

Why do you think it could crash? `ref MemoryMarshal.GetReference` handles null just fine.

This ` if (_entryBuffer == null) return;` should be deleted, and null check should be added to `_entryBuffer.Length` a few lines below.

Notice that when this method is called there is always `MoveNext()` next on the stack that has a dummy `fixed (byte* _ = _entryBuffer)` to keep the buffer pinned. Ideally, the pointer to the buffer pinned by `MoveNext()` would be passed along and this would just use that pointer instead of doing another redundant fixed. Unfortunately, it cannot be done easily given how to code is factored.

Reply:
>  I refactored a bit FindNextEntry for cases where it's possible to pass a pointer.

I am not sure whether this refactoring is an improvement. It would be an improvement if it is always possible to pass a pointer. Performance-wise, it is probably a tiny loss (the redundant `fixed` is very cheap).